examples/2d/sprite_animation.rs

```diff
@@ -0,0 +1,124 @@
+//! Animates a sprite in response to a keyboard event.
+//!
+//! See `sprite_sheet.rs` for an example where the sprite animation loops indefinitely.
+
+use std::time::Duration;
+
+use bevy::input::common_conditions::input_just_pressed;
+use bevy::prelude::*;
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins.set(ImagePlugin::default_nearest())) // prevents blurry sprites
+        .add_systems(Startup, setup)
+        .add_systems(Update, execute_animations)
+        // press the left arrow key to animate the left sprite
+        .add_systems(
+            Update,
+            trigger_animation::<LeftSprite>.run_if(input_just_pressed(KeyCode::ArrowLeft)),
+        )
+        // press the right arrow key to animate the right sprite
+        .add_systems(
+            Update,
+            trigger_animation::<RightSprite>.run_if(input_just_pressed(KeyCode::ArrowRight)),
+        )
```

Comment:
We should also add a basic ui to describe this keys for the wasm example

Reply:
<img width="1272" alt="Screenshot 2024-04-23 at 12 40 55" src="https://github.com/bevyengine/bevy/assets/29188350/b491b7aa-ef1c-451d-91bd-2b41c05c7e23">

Done!